Added user:group to the file footer.

rickygzz opened the pull request:
Hello,

Here with my two cents. This is my first commit.

Adding user:group to the file footer

![image](https://github.com/user-attachments/assets/01c9435c-3216-46a4-ab62-8b0dd03185af)

![image](https://github.com/user-attachments/assets/92e1df36-4dfa-4d94-a183-d8b7336c9a18)

jelly commented:
Thanks for the pull request, I've just re-read the issue and @garrett wrote that it should be between the modified string and permissions. This makes sense as it is the same as our list view.

jelly commented on src/files-footer-detail.tsx:
```diff
@@ -122,3 +128,41 @@ export const FilesFooterDetail = ({
+            isInline
+            component="pre"
+        >
+            {file.user}:{file.group}
```
This needs to be a little more complicated and simplify showing only the user when the user and group are equal. We do this already in `FileOwnership` in the files-card-body.tsx

garrett replied:
I agree with this. (Thanks for mentioning it!)

jelly commented on src/files-footer-detail.tsx:
```diff
@@ -8,7 +8,8 @@ import { Popover } from "@patternfly/react-core/dist/esm/components/Popover";
 import { Tooltip } from "@patternfly/react-core/dist/esm/components/Tooltip";
 
-import cockpit from "cockpit";
+import cockpit, { file } from "cockpit";
```
this file import should not be needed (or even resolve)

garrett reviewed (dismissed):
Looking good.

However, a few issues that @jelly already pointed out (and that I agree with):

- Placement: it should be right before permissions (and after modified time)
- Collapsing consistency: Shrink user:group down to just user when user == group, like we do in the list.

Once those are implemented, I'm happy with this when @jelly is happy with this.

rickygzz commented:
Thank you for your kind review!

1. Rearranged user:group to be in between the modified string and permissions
2. Added shrinking logic when user==group
3. Removed unneeded file import
4. Rebased code

Please  @jelly let me know your comments

jelly commented on src/files-footer-detail.tsx:
```diff
@@ -118,7 +123,46 @@ export const FilesFooterDetail = ({
+            >
+                <DescriptionListGroup key="user">
+                    <DescriptionListTerm>
+                        User
```
This needs to be translated

```suggestion
{_("User")}
```

jelly reviewed (changes requested):
Are you interested in adding tests, otherwise I can help or write them.

rickygzz commented:
@jelly 

Sorry for the delay. I updated the translatable texts and rebased my commits.

I spent couple hours trying to figure out the tests in this project but I guess it will take me longer to understand them.

```console
$ make check
tools/node-modules make_package_lock_json
bots/image-customize --no-network --fresh \
         \
        --upload cockpit-files-node-1.tar.xz:/var/tmp/ --build cockpit-files-1.tar.xz \
        --script /home/rocky/repos/cockpit-files/test/vm.install fedora-41
Traceback (most recent call last):
  File "/home/rocky/repos/cockpit-files/bots/image-customize", line 26, in <module>
    from machine import testvm
  File "/home/rocky/repos/cockpit-files/bots/machine/testvm.py", line 28, in <module>
    from lib.directories import get_images_data_dir
  File "/home/rocky/repos/cockpit-files/bots/lib/directories.py", line 25, in <module>
    def xdg_home(subdir: str, envvar: str, *components: str, override: str | None = None) -> str:
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
make: *** [Makefile:187: /home/rocky/repos/cockpit-files/test/images/fedora-41] Error 1
```

I guess the test code should look something like

```py
        # Test ownership
        m.execute("chown admin:root /home/admin/newdir")
        b.wait_in_text(".files-footer-info .pf-m-link", "admin:root")
        m.execute("chown admin:admin /home/admin/newdir")
        b.wait_in_text(".files-footer-info .pf-m-link", "admin")
        # Also check popover
        b.click(".files-footer-info .pf-m-link")
        b.wait_in_text(".pf-v5-c-popover dl div:nth-child(1) > dd", "admin")
        b.wait_in_text(".pf-v5-c-popover dl div:nth-child(2) > dd", "admin")
        b.click(".pf-v5-c-popover__close > button")
        b.wait_not_present(".pf-v5-c-popover")
```

And I am guessing I will have to tweak the .pf-m-link "path" since now we have two: one for the ownership link and one for the permissions link. But to play around first I need to figure out how to run the tests properly.

I guess it would be faster if you write them down for me, what do you think?

jelly commented:
Taken the liberty to squash your commits, apply some eslint fixes and add tests. This should be good to go now.

martinpitt commented on test/check-application:
```diff
@@ -333,6 +333,7 @@ class TestFiles(testlib.MachineCase):
         b.wait_in_text("#files-footer-permissions", "rwx r-x r-x")
         m.execute("chmod 700 /home/admin/newdir")
         b.wait_in_text("#files-footer-permissions", "rwx --- ---")
+        b.wait_in_text("#files-footer-owner", "root")
```
Let's make this stricter to `wait_text()` -- we don't expect a "root:root" (or even other values) here.

martinpitt commented on test/check-application:
```diff
@@ -342,6 +343,22 @@ class TestFiles(testlib.MachineCase):
+
+        # Change group is shown in popover
+        m.execute("chown root:admin /home/admin/newdir")
+        b.wait_in_text("#files-footer-owner", "root:admin")
```
can also be wait_text()

martinpitt reviewed (dismissed):
Thanks! Some suggestions to make the tests stricter, but LGTM!
